frontend/wallet/src/common/CodaCliSettings.re

```diff
@@ -0,0 +1,46 @@
+// TODO: Put this somewhere and implement properly
+module ProvingKey = {
+  // path to a file
+  type t = string;
+};
+
+module WithKey = {
+  type t('a) =
+    | NoKey
+    | Key(ProvingKey.t, 'a);
+};
+
+module General = {
+  type t = {testnet: Url.t};
+};
+
+module Wallets = {
+  type t = WithKey.t(option(PublicKey.t));
```

Comment:
why do we need a withkey and an optional?

Reply:
3 states for your staking option in the wallets subcategory:
1. NoKey -- you haven't downloaded the staking proving key
2. Key(provingKey, None) -- you have downloaded the key, but you are not staking with any wallet
3. Key(provingKey, Some(pubkey)) -- you have downloaded the key and enabled staking on this particular wallet denoted by pubkey

Reply:
Can't hasKey just be a boolean then? And then Key can just be an optional?

Reply:
b/c technically you could have a key set for staking but the proving key could be corrupted

Reply:
we won't be able to detect if the key is corrupted unless we add some sort of checksumming thing into it

But perhaps it could be deleted from the filesystem or something -- okay I'm convinced